### ISSUE TITLE
feat(GoogleMap): Add google controls integration

### ISSUE THREAD

**Issue by labeled** [angular/components#22526]
#### Feature Description

The google maps api has the ability to add controls the map. It would be nice to have an 'angular' way to push components into the controls api. See https://developers.google.com/maps/documentation/javascript/examples/control-custom for examples.

#### Use Case

This would allow developers capability to add buttons like location to the existing maps instead of overlaying a custom elements over the map.

**Comment by b00ling** [comment]
> #### Feature Description
> The google maps api has the ability to add controls the map. It would be nice to have an 'angular' way to push components into the controls api. See https://developers.google.com/maps/documentation/javascript/examples/control-custom for examples.


Hi,
is there a 'not only angular' way at the moment?

**Comment by labeled** [comment]
Yeah currently you create an html object using javascript and inject it
into the map using something like this.

map.controls[google.maps.ControlPosition.TOP_CENTER].push(centerControlDiv);


It would be nice to use an angular component and have it used on the map
instead of vanilla javascript.

*Thanks,*

*___________________________*
*Travis Peterson*
*Full Stack Developer*
Altec Industries
1255 Port Terminal Dr
Duluth MN, 55802
(c) 763-442-0747
(w) 218-733-1215

**Comment by ynoval** [comment]
We need to add a Layers Manager as you have when works with Google Earth Engine Code. We did :

//Just HTML element 
const containerEl = document.getElementById('containerLayers');

// Create Element (Angular Component MapLayerComponent with selector app-map-layers)
const layersElement: NgElement & WithProperties<MapLayersComponent> = document.createElement(
          'app-map-layers'
        ) as any;

// Listen ChangeLayer event (when user select/ unselect a layer in the component)
        layersElement.addEventListener('changeLayer', (event: any) => {
          if (event.detail.action === 'SHOW') {
            this.showLayer(event.detail.layer);
          } else {
            this.hideLayer(event.detail.layer);
          }
        });

        // Pass layers to the component
        layersElement.mapLayers = this.mapLayers;
        // Add to the DOM (container)
        containerEl.innerHTML = '';
        containerEl.appendChild(layersElement);
        this.mapInstance.googleMap.controls[google.maps.ControlPosition.TOP_RIGHT].push(containerEl);

Hope this can help.

**Comment by jermowery** [comment]
I would also love to see this added. The current process is clunky, especially if you have an angular component you are trying to use as a control you need do dynamic component creation and some magic with the Renderer
